src/groupByKey.ts

```diff
@@ -2,6 +2,13 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  return items.reduce((groups: GroupsMap<T>, item: T) => {
+    const groupKey = item[key] as string;
```

Comment:
Type assertion using 'as string' can be unsafe if 'key' does not actually refer to a string property. Consider adding a type guard to ensure 'key' is a string property.